File: components/audit/src/main/java/org/cloudfoundry/credhub/audit/HttpUtils.java
package org.cloudfoundry.credhub.audit;

import org.springframework.http.HttpStatus;

final public class HttpUtils {
  private HttpUtils() {
    super();
  }

  public static String getResultCode(final int statusCode) {
    if (statusCode < HttpStatus.OK.value()) {
      return "info";
    } else if (statusCode < HttpStatus.MULTIPLE_CHOICES.value()) {
      return "success";
    } else if (statusCode < HttpStatus.BAD_REQUEST.value()) {
      return "redirect";
    } else if (statusCode < HttpStatus.INTERNAL_SERVER_ERROR.value()) {
      return "clientError";
    } else {
      return "serverError";
    }
  }
}

File: components/audit/src/main/kotlin/org/cloudfoundry/credhub/audit/HttpUtils.kt
package org.cloudfoundry.credhub.audit

import org.springframework.http.HttpStatus

class HttpUtils private constructor() {
    companion object {
        fun getResultCode(statusCode: Int): String =
            if (statusCode < HttpStatus.OK.value()) {
                "info"
            } else if (statusCode < HttpStatus.MULTIPLE_CHOICES.value()) {
                "success"
            } else if (statusCode < HttpStatus.BAD_REQUEST.value()) {
                "redirect"
            } else if (statusCode < HttpStatus.INTERNAL_SERVER_ERROR.value()) {
                "clientError"
            } else {
                "serverError"
            }
    }
}
